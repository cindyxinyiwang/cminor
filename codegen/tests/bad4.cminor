// function type error

hello1: function integer () = {
	print "hello world";
}

x: integer = 1;
y: boolean = true;
y1: boolean = false;

// return type
random: function array [3] integer (x: integer, y: integer) = {
	z: integer = x+y;
	a: array [3] integer = {1, 2,3};
	return z;
}

func: function array [10] boolean (m: boolean, n: string, y: integer) = {
	print m, n, y;

	y = x;
	m = y1 || x==y;

	// param error
	random();
	random(true, 1);
	return random(1, 2, 3);
}

proto: function void (i: integer, j: boolean, k: string, p: char);

// proto error
useProto: function void (p: integer, q:integer) = {
	x = proto(p, true, "hello", 'h');
	func(true, "hello", 1);

	proto(1, true, 'h');
	proto(1, 1, "hello","h");
}
