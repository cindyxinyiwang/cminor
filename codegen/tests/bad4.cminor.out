x resovles to param 0
y resovles to param 1
z resovles to local 0
m resovles to param 0
n resovles to param 1
y resovles to param 2
y resovles to param 2
x resovles to global x
m resovles to param 0
y1 resovles to global y1
x resovles to global x
y resovles to param 2
random resovles to global random
random resovles to global random
random resovles to global random
x resovles to global x
p resovles to param 0
proto resovles to global proto
func resovles to global func
proto resovles to global proto
proto resovles to global proto
type error: cannot return a integer in a function(random) that returns array [3] integer
type error: not enough variables to function call random
type error: cannot pass variable of type boolean to function random
type error: pass more variables to functions call random than needed
type error: cannot return a array [3] integer in a function(func) that returns array [10] boolean
type error: cannot assign void to a integer
type error: cannot pass variable of type character to function proto
type error: not enough variables to function call proto
type error: cannot pass variable of type integer to function proto
type error: cannot pass variable of type string to function proto
